app/lib/bk/compat/parsers/bitrise/workflows.rb

```diff
@@ -7,12 +7,15 @@ class Bitrise
       private
 
       def parse_workflow(wf_name, wf_config)
+        agents = { stack: wf_config['stack'] } if wf_config.include?('stack')
         # Extend for other types of steps - approvals/waits for example.
-        BK::Compat::CommandStep.new(label: wf_name, key: wf_name).tap do |cmd_step|
+        BK::Compat::CommandStep.new(label: wf_name, key: wf_name, agents: agents || {}).tap do |cmd_step|
           wf_config['steps'].each do |step|
             step_key, step_config = step.first
-            step_inputs = step_config['inputs'].reduce({}, :merge)
-            cmd_step << load_workflow_step(step_key, step_inputs)
+            step_inputs = (step_config['inputs'] || {}).reduce({}, :merge)
```

Comment:
the default should not be necessary as it was working until now, or am I missing anything?